packages/qwik/src/qwikloader.ts

```diff
@@ -60,15 +62,20 @@ export const qwikLoader = (doc: Document, hasInitialized?: number) => {
     if (element.hasAttribute('preventdefault:' + eventName)) {
       ev.preventDefault();
     }
-    if (element.hasAttribute('stoppropagation:' + eventName)) {
-      ev.stopPropagation();
-    }
     const ctx = (element as any)['_qc_'] as QContext | undefined;
     const relevantListeners = ctx && ctx.li.filter((li) => li[0] === attrName);
     if (relevantListeners && relevantListeners.length > 0) {
       for (const listener of relevantListeners) {
         // listener[1] holds the QRL
-        await listener[1].getFn([element, ev], () => element[isConnected])(ev, element);
+        const results = listener[1].getFn([element, ev], () => element[isConnected])(ev, element);
+        const cancelBubble = ev.cancelBubble;
+        if (isPromise(results)) {
+          await results;
+        }
+        // forcing async with await resets ev.cancelBubble to false
+        if (cancelBubble) {
+          ev.stopPropagation();
```

Comment:
Shouldn't this be called before awaiting?

Reply:
> Shouldn't this be called before awaiting?

Ah yes, I don't think we want any possible delay here, so something like:

```tsx
const cancelBubble = ev.cancelBubble;
if (cancelBubble) {
  ev.stopPropagation();
}
if (isPromise(results)) {
  await results;
}
```

does that look right @wmertens ?